### DISCUSSION TITLE
Question: what to other people "see" from other servers/federations?

### DISCUSSION THREAD

**Discussion by geecrumb** [balzack/databag#103] [Q&A]
I still dont know what other people "could" see from my server. I understand that my server is somehow connected to a "master hive" and I could search for different servers in the contacts (if I know its name). But if someone knows the name of my server, are they able to see:

1. Profile picture of my users?
2. See or read topics somehow (?) when its not encrypted?
3. Can randomly connect to other users on my server?

What happens if I dont set up the federation in my admin settings? Will my users still be able to search other users on my server (so this federation thing is a must) or is it just to be able to be found from other servers?

Thanks :)

**Comment by balzack** [comment]
There is no master hive; each node operates independently with no built in knowledge of another. If someone knows the name of your server, there are only four public endpoints:

   /account/listing - returns the list of profiles on the server marked as searchable. The profile is { image, name, location, description }
   /account/listing/{guid}/image - returns the profile image of a specified account if it's marked as searchable

   /contact/openMessage - request connection to an account, payload includes signed data identifying sender
   /contact/closeMessage - close connection from an account, payload includes signed data identifying sender

To your points:
1. The profile picture is only publicly available if the account marks itself as searchable. If the account is not marked as searchable, nothing is returned publicly.
2. No, to view a topic, a person must connect to the account hosting the topic, and that topic must be shared with the person. Encrypting the topic is mainly beneficial is maintaining privacy from people with unrestricted access to the server. A system admin for example can open up the server database via command line and view the messages. A sealed or end-to-end encrypted topic will still be unreadable with unrestricted access to the server.
3. Connecting to a user currently requires sending a connection request. That connection request needs to be approved by the user before any access to topics is granted. I have thoughts on how to restrict the initial connection request, but nothing implemented yet.

If you don't federate, the accounts should still be able to communicate between each other without restriction. Note however, there is no option to block federation. If your server is accessible, an account on a different server can still send a connection request to an account on your server.
